torch/csrc/autograd/engine.cpp

```diff
@@ -93,6 +93,23 @@ struct ReadyQueue {
   size_t size() const;
 };
 
+// This shared_ptr is a thread local pointer to the local ready queue per thread
+// We colocate each device (i.e. CUDA, XLA) a separate thread other than CPU,
+// see Note [Allocating GPUs to autograd threads] each device thread should have
+// its own ReadyQueue that initialized in thread_init and used as the queue for
+// executing tasks. These local_ready_queues for device threads are also memorized
+// in the Engine to performance cross device training (i.e. CPU to GPU, XLA, etc.)
+//
+// For cpu threads, each thread will also have its own ReadyQueue per thread, and
+// it is memorized in the GraphTask to perform cross device training. (i.e. GPU to
+// CPU via variable.cpu(), etc.)
+//
+// For reentrant backward calls, if we spawn new thread from the current thread
+// while we reach the depth (current_depth >= total_depth), the new thread will
+// just reuse the same readyqueue with the parent thread for a mild performance
+// improvement. see Note [Reentrant backwards] for more details.
+static thread_local std::shared_ptr<ReadyQueue> local_ready_queue = nullptr;
```

Comment:
What is the benefit of this? Only to avoid the creation of a ready queue when doing re-entrant autograd?